lib/pipeline.js

```diff
@@ -75,30 +81,32 @@ class PipelineModel extends BaseModel {
             toUnarchive: []
         };
         const existingPRs = existingJobs.filter(j => j.isPR());   // list of PRs according to SD
-        const existingPRsNames = existingPRs.map(j => j.name);
         const openedPRsNames = openedPRs.map(j => j.name);
         const openedPRsRef = openedPRs.map(j => j.ref);
 
         existingPRs.forEach((job) => {
+            // getting PR and number e.g. PR-1
+            const prName = job.name.match(PR_JOB_NAME)[PR_NAME_GROUP.pr];
+
             // if PR is closed, add it to archive list
-            if (openedPRsNames.indexOf(job.name) < 0 && job.archived === false) {
+            if (!openedPRsNames.includes(prName) && !job.archived) {
                 jobList.toArchive.push(job);
             }
         });
 
         openedPRsNames.forEach((name, i) => {
-            const index = existingPRsNames.indexOf(name);
-
-            if (index < 0) {
-                // if opened PR is not in the list of existingPRs, create it
-                jobList.toCreate.push({ name, ref: openedPRsRef[i] });
-            } else {
-                const job = existingPRs[index];
+            const matchedPRs = existingPRs.filter(job => job.name.startsWith(name));
 
-                // if opened PR was previously archived, unarchive it
+            // if opened PR was previously archived, unarchive it
+            matchedPRs.forEach((job) => {
                 if (job.archived) {
-                    jobList.toUnarchive.push(existingPRs[index]);
+                    jobList.toUnarchive.push(job);
                 }
+            });
+
+            // if opened PR is not in the list of existingPRs, create it
+            if (matchedPRs.length === 0) {
+                jobList.toCreate.push({ name, ref: openedPRsRef[i].ref });
```

Comment:
I don't think you need `.ref` here. `openedPRsRef` should already just contain `job.ref`.